crates/bin/cairo0_run/src/main.rs

```diff
@@ -0,0 +1,87 @@
+use cairo_vm::cairo_run::CairoRunConfig;
+use cairo_vm::types::layout_name::LayoutName;
+use cairo_vm::types::program::Program;
+use cairo_vm::vm::errors::vm_exception::VmException;
+use cairo_vm::vm::runners::cairo_runner::CairoRunner;
+use cairo_vm::Felt252;
+use clap::Parser;
+use starknet_os::error::SnOsError;
+use starknet_os::hints;
+use starknet_os::starknet::starknet_storage::{CommitmentInfo, CommitmentInfoError, PerContractStorage};
+use starknet_os::starkware_utils::commitment_tree::base_types::TreeIndex;
+
+#[derive(Parser, Debug)]
+struct Args {
+    #[arg(long)]
+    input: String,
+}
+
+fn init_logging() {
+    env_logger::builder()
+        .filter_level(log::LevelFilter::Trace)
+        .format_timestamp(None)
+        .try_init()
+        .expect("Failed to configure env_logger");
+}
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    init_logging();
+
+    let args = Args::parse();
+
+    let program_bytes = std::fs::read_to_string(args.input).expect("Failed to read input file");
+
+    let layout = LayoutName::all_cairo;
+
+    // Init CairoRunConfig
+    let cairo_run_config = CairoRunConfig { layout, relocate_mem: true, trace_enabled: true, ..Default::default() };
+    let allow_missing_builtins = cairo_run_config.allow_missing_builtins.unwrap_or(false);
+
+    // Load the Starknet OS Program
+    let program = Program::from_bytes(program_bytes.as_bytes(), Some(cairo_run_config.entrypoint))
+        .map_err(|e| SnOsError::Runner(e.into()))?;
+
+    // Init cairo runner
+    let mut cairo_runner = CairoRunner::new(
+        &program,
+        cairo_run_config.layout,
+        cairo_run_config.proof_mode,
+        cairo_run_config.trace_enabled,
+    )
+    .map_err(|e| SnOsError::Runner(e.into()))?;
+
+    // Init the Cairo VM
+    let end = cairo_runner.initialize(allow_missing_builtins).map_err(|e| SnOsError::Runner(e.into()))?;
+
+    // Run the Cairo VM
+    let mut hint_processor = hints::SnosHintProcessor::<DummyPCS>::default();
```

Comment:
Why are we using a Dummy storage?

Reply:
Because this isn't meant to do any storage ops, it's really just meant to run cairo0 programs. I suppose some OS features could be supported, but that should be implemented when needed...